src/main/kotlin/com/codingchallenge/transactionanalyser/TransactionLoader.kt

```diff
@@ -0,0 +1,35 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.Transaction
+import com.codingchallenge.transactionanalyser.domain.TransactionBuilder
+import java.io.BufferedReader
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.util.stream.Collectors
+
+class TransactionLoader(private val input: InputStream) {
```

Comment:
I hope you can see here a good example of the Kotlin language itself being **immutability** as part of its design. It is a big thing we should always be thinking of, and consider whether you are writing Java style code in Kotlin, or are you truly writing immutable Kotlin code in Kotlin. (or are you just writing Java style to be compatible with Java libraries, hehe).

What I mean by this specifically is that your `TransactionLoader` is taking in a value in its constructor and it is a `val`. We aren't making getters and setters where `input` could be modified at any point later. It gives us more confidence in `TransactionLoader` to have a predictable output from what you pass in. Of course that is not always the case but notice the language is encouraging us to work this way.

Reply:
Seems like Kotlin has the `final` keyword baked into `val`.